### PR TITLE
Implement finally()

### PR DESCRIPTION
**Pull request by Avaq**
This commit will add a new function:

```js
.finally :: Future a b -> Future a c -> Future a c
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -350,6 +350,36 @@ Future.reject('it broke')
 //> Left('it broke')
 ```
 
+#### finally
+##### `#finally :: Future a b ~> Future a c -> Future a b`
+##### `.finally :: Future a b -> Future a c -> Future a c`
```
Comment (davidchambers) [inline README.md]: Are these return types different intentionally?

Comment (Avaq) [inline README.md]: The functional API takes the "data" last, whereas in the method API `this` *is* the data.

So:

```js
computation.finally(finallyComputation)
==
Future.finally(finallyComputation, computation)
```

The different output type is meant to draw attention to that fact.

Comment (Avaq) [inline README.md]: Perhaps it'd be better to leave the output type the same `Future a b` and switch `b` and `c` between the first two types.

Comment (davidchambers) [inline README.md]: I like that idea! Having the return types match would make it clear that the arguments are flipped (and that there's no typo).

Comment (Avaq) [inline README.md]: :zap:

File: README.md [hunk 2]
```diff
@@ -350,6 +350,36 @@ Future.reject('it broke')
 //> Left('it broke')
 ```
 
+#### finally
+##### `#finally :: Future a b ~> Future a c -> Future a b`
+##### `.finally :: Future a c -> Future a b -> Future a b`
+
+Run a second Future after the first ends, despite of it rejecting or resolving.
```
Comment (davidchambers) [inline README.md]: This doesn't quite make sense. I suggest:

_Run a second Future after the first ends (successfully or unsuccessfully)._

Comment (Avaq) [inline README.md]: Good idea

**Pull request merged**